src/subwindows/generalSettingsDialog.cpp

```diff
@@ -95,37 +101,42 @@ void GeneralSettingsDialog::updateForm() {
     // GB modified begin
     // NOTE: thiw line below somehow does not work (always sets the index 0 element of the combobox)
     // imageWriterFormatBox->setCurrentText(imageWriterFormat);
-    if(imageWriterFormat == "tiff")
-        imageWriterFormatBox->setCurrentIndex(0);
-    else if(imageWriterFormat == "jpg")
+    if(imageWriterFormat == "jpg")
         imageWriterFormatBox->setCurrentIndex(1);
     else if(imageWriterFormat == "png")
         imageWriterFormatBox->setCurrentIndex(2);
     else if(imageWriterFormat == "bmp")
         imageWriterFormatBox->setCurrentIndex(3);
+    else // if(imageWriterFormat == "tiff")
+        imageWriterFormatBox->setCurrentIndex(0);
     //qDebug() << imageWriterFormat << "\n";
 
     if(imageWriterDataRule == "ask")
         imageWriterDataRuleBox->setCurrentIndex(0);
     else if(imageWriterDataRule == "append")
         imageWriterDataRuleBox->setCurrentIndex(1);
-    else if(imageWriterDataRule == "new")
+    else // if(imageWriterDataRule == "new")
         imageWriterDataRuleBox->setCurrentIndex(2);
 
-    //delimiterBox->setCurrentText(delimiterToUse);
-    if(delimiterToUse == ",")
-        delimiterBox->setCurrentIndex(0);
-    else if(delimiterToUse == ";")
-        delimiterBox->setCurrentIndex(1);
-    else if(delimiterToUse == "\t")
-        delimiterBox->setCurrentIndex(2);
-    //qDebug() << delimiterToUse << "\n";
+    //dataWriterDelimiterBox->setCurrentText(delimiterToUse);
+    if(dataWriterDelimiter == ";")
+        dataWriterDelimiterBox->setCurrentIndex(1);
+    else if(dataWriterDelimiter == "\t")
+        dataWriterDelimiterBox->setCurrentIndex(2);
+    else //if(dataWriterDelimiter == ",")
+        dataWriterDelimiterBox->setCurrentIndex(0);
+    qDebug() << "Data writer delimiter read as: " << dataWriterDelimiter << "\n";
+
+    if(dataWriterDataStructure == "PupilEXT-0-1-1")
```

Comment:
This should maybe use the index instead of the string?

Reply:
Left like this for now